src/lib/banal/Struct.cpp

```diff
@@ -1638,7 +1638,14 @@ doFunctionList(WorkEnv & env, FileInfo * finfo, GroupInfo * ginfo, bool fullGaps
 #endif
   }
 
-  if (cuda_arch == 0) {
+  // computeGaps relies Block::start() and Block::end() to have
+  // the correct address range for a block to detect gaps.
+  // We have subclass GPUBlock inheriting ParseAPI::Block.
+  // However, currently subclass of ParseAPI::Block cannot update
+  // block end, which leads to Block::end() == Block::start() for
+  // GPUBlock objects.
+  // Therefore, we cannot invoke gap computation for GPU binaries
+  if (cuda_arch == 0 && intel_gpu_arch == 0) {
```

Comment:
isn’t it the case that blocks have a proper block end address and and meet the requirements for gap parsing?  
if we don’t want to do gap parsing for gpu binaries, that’s a
choice. however are the comments above that say we can’t still right?

Reply:
@jmellorcrummey You are correct that the comments are not long valid. I am going to update the comment. I don't think we should invoke gap computation for GPU code even if we have presumably correct  block boundary since we have never tested its impact.

Reply:
@jmellorcrummey Comments updated.